simpleio.py

```diff
@@ -31,9 +31,73 @@
 """
 
 import digitalio
-import math
+import time
+
 from neopixel_write import neopixel_write
 
+def shiftIn(dataPin, clock, msb_first=True):
+    """
+    Shifts in a byte of data one bit at a time. Starts from either the LSB or
+    MSB.
+
+    :param dataPin: pin on which to input each bit
+    :param clock: toggles to signal dataPin reads
+    :param msb_first: order to shift bits (least significant or most significant bit first)
+    :return: returns the value read
+    :rtype: int
+    """
+
+    value = 0
+    i = 0
+
+    for i in range(0, 8):
+        clock.value = True
+        if msb_first == False:
```

Comment:
I'd reverse this:

```python
if msb_first:
  value |= ((dataPin.value) << (7-i))
else:
  value |= ((dataPin.value) << i)
```

If tests that are positive are generally easier to read.